### PR TITLE
Support skin tone modifiers

### PR DESCRIPTION
**Pull request by dgraham**
Adds a `tone` attribute and property to modify the skin tone of the displayed emoji character. This allows a default emoji rendered in the initial document to be customized via script after page load.

```html
<g-emoji>👋</g-emoji>
<!-- becomes -->
<g-emoji tone="3">👋🏽</g-emoji>
```

References:
- https://unicode.org/emoji/charts/full-emoji-modifiers.html
- https://en.wikipedia.org/wiki/Miscellaneous_Symbols_and_Pictographs#Emoji_modifiers

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -13,6 +14,14 @@ class GEmojiElement extends HTMLElement {
     }
   }
 
+  get tone(): number {
+    return this.hasAttribute('tone') ? parseInt(this.getAttribute('tone'), 10) : 0
```
Comment (muan) [inline index.js]: Considering this as an enumerated attribute, I think the getter should always return valid values. Similar to how [the `autocomplete` attribute behaves](https://html.spec.whatwg.org/#dom-fe-autocomplete): 

> The autocomplete IDL attribute, on getting, must return the element's IDL-exposed autofill value, and on setting, must reflect the content attribute of the same name.

```suggestion
    const tone = parseInt(this.getAttribute('tone'), 10)
    return MODIFIERS.includes(tone) ? tone : 0
```

File: index.js [hunk 2]
```diff
@@ -21,6 +30,65 @@ class GEmojiElement extends HTMLElement {
...
   }
+
+  static get observedAttributes(): Array<string> {
+    return ['tone']
+  }
+
+  attributeChangedCallback(name: string, oldValue: string, newValue: string) {
+    switch (name) {
+      case 'tone':
+        updateTone(this, parseInt(newValue, 10))
```
Comment (muan) [inline index.js]: We can avoid `parseInt` in multiple places by relying on the getter.

```suggestion
        updateTone(this)
```

File: index.js [hunk 3]
```diff
@@ -21,6 +30,65 @@ class GEmojiElement extends HTMLElement {
...
+        break
+    }
+  }
+}
+
+function updateTone(el: GEmojiElement, modifier: number) {
+  if (el.image) return
+  if (!isModifiable(el.textContent)) return
+
+  const point = toneModifier(modifier)
```
Comment (muan) [inline index.js]:
```suggestion
  const point = toneModifier(el.tone)
```

File: index.js [hunk 4]
```diff
@@ -13,6 +14,14 @@ class GEmojiElement extends HTMLElement {
     }
   }
 
+  get tone(): number {
+    return this.hasAttribute('tone') ? parseInt(this.getAttribute('tone'), 10) : 0
+  }
+
+  set tone(modifier: number) {
```
Comment (muan) [inline index.js]: Perhaps an overkill 🙈  but we can potentially tighten this up a little with a `MODIFIERS` type so `emoji.tone = 999` fails type check.

```suggestion
  set tone(modifier: MODIFIERS) {
```

File: script/generate-emoji-modifiers [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+// You MUST commit any modifications to this file.
+
+/* @flow strict */
+
+const points = new Set([
+  ${[...points].sort().join(',')}
+])
+
+export function isModifiable(emoji: string): boolean {
+  return points.has(emoji.codePointAt(0))
```
Comment (muan) [inline script/generate-emoji-modifiers]: Ooof e188ccc reminded me that https://blog.emojipedia.org/service-dog-deaf-person-couples-added-to-2019-emoji-list/#aremixedracefamiliesincluded is a thing.

> Unicode already provides a way for vendors to support mixed-race families, should any wish to do so. These are supported on Windows, and a subset where all family members share a skin tone are also supported on Facebook.
>
> **As of Emoji 12.0, no new family combinations are listed by Unicode, potentially due to the number of combinations required when three or four family members are included.**

They are marked as not modifiable in `emojilib`.

https://github.com/muan/emojilib/blob/abd4322e610a0dd4b38fc7f494c49656791e2263/emojis.json#L1736-L1741

If we want to support this, we'd need to check and modify based on each emoji. If we don't (which I am leaning towards), we'd need to check modifiable based on fully composited emoji.

File: index.d.ts [hunk 6]
```diff
@@ -1,5 +1,9 @@
+type SkinTone = 0 | 1 | 2 | 3 | 4 | 5
+
 export default class GEmojiElement extends HTMLElement {
   image?: HTMLImageElement
+  tone: SkinTone
+  tones: Array<SkinTone>
```
[the remark this answers is not in the document]
Comment (dgraham) [reply index.d.ts]: 2fafffc7

**Pull request merged**